.buildkite/scripts/benchmark/main.sh

```diff
@@ -107,6 +107,9 @@ pull_images() {
     docker pull "docker.elastic.co/logstash/logstash:$LS_VERSION"
   else
     LS_VERSION=$( curl --retry-all-errors --retry 5 --retry-delay 1 -s https://artifacts-api.elastic.co/v1/versions | jq -r ".versions[-1]" )
+    if [[ ! $LS_VERSION == *-SNAPSHOT ]]; then
+      LS_VERSION="${LS_VERSION}-SNAPSHOT"
+    fi
     BUILD_ID=$( curl --retry-all-errors --retry 5 --retry-delay 1 -s https://artifacts-api.elastic.co/v1/branches/master/builds | jq -r ".builds[0]" )
```

Comment:
Possibly out-of-scope, but the artifacts API calls out getting the appropriate build id with `https://artifacts-api.elastic.co/v1/versions/$VERSION/builds/latest` ([link](https://github.com/elastic/release-eng/tree/main/containers/artifacts-api#quickstart))

As of this writing, we're getting a slightly-older build by querying off `/v1/branches/master/builds` (`8.15.0-90bfa704`, which has `last-modified: Wed, 03 Jul 2024 14:08:42 GMT`) instead of `/v1/versions/:version/builds/latest` (`8.15.0-807b1430`, which has `last-modified: Mon, 08 Jul 2024 16:10:23 GMT`).

My best guess is that when we cut the branch for 8.15 at feature freeze, we stopped building 8.15 artifacts from `main`, so the `master` no longer has new snapshots as of that feature freeze.

> ~~~ suggestion
>     BUILD_ID=$(curl --retry-all-errors --retry 5 --retry-delay 1 -s "https://artifacts-api.elastic.co/v1/versions/${LS_VERSION}/builds/latest" | jq -re '.build.build_id')
> ~~~

Reply:
Great catch! It is so weird that `https://artifacts-api.elastic.co/v1/branches/master/builds` gives me `8.15.0-90bfa704` with last modified time `Mon, 08 Jul 2024 20:03:37 UTC`. By comparing DRA snapshot SHA, I believe you are right that `/v1/versions/8.15/builds/latest` gives most updated version and the build pipeline stops pushing build id to `/v1/branches/master/builds`.